### DISCUSSION TITLE
How does PA -> P conversion work?

### DISCUSSION THREAD

**Discussion by Yay295** [python-pillow/Pillow#6506] [Q&A] [answered]
There is no PA to P function in [Convert.c](https://github.com/python-pillow/Pillow/blob/main/src/libImaging/Convert.c). I tried following the code but I must have missed something because I didn't see where this conversion happens.

**Comment by radarhere** [comment, marked as answer]
You are correct that PA is not directly converted to P in the C layer at the moment. When support is missing for a direct conversion, the Python layer instead converts first to the base mode, and then to the destination mode.

https://github.com/python-pillow/Pillow/blob/87ecd01fc042bb67b11bfcff3db99fc2b61e702d/src/PIL/Image.py#L1034-L1040

So it goes PA to RGB, and then RGB to P.

**Reply by radarhere** [reply]
With #6509, this answer has changed - there is now a [pa2p function in Convert.c.](https://github.com/python-pillow/Pillow/blob/d80e93e4353210ddb3081e682971a073af4dc6d7/src/libImaging/Convert.c#L1221)
